### PR TITLE
linh-add-customer

### LINKED ISSUE #67: Implment add customer page

### PR DESCRIPTION
**Pull request by linhlm0210**

### CODE REVIEW

File: src/main/java/io/hardingadonis/saledock/controller/management/customer/AddCustomerServlet.java [hunk 1]
```diff
@@ -1,20 +1,57 @@
...
+        requestDispatcher.forward(request, response);
+
     }
 
     @Override
     protected void doPost(HttpServletRequest request, HttpServletResponse response)
             throws ServletException, IOException {
+
+       
+            CustomerDAOImpl dao = new CustomerDAOImpl();
```
Comment (hardingadonis) [inline src/main/java/io/hardingadonis/saledock/controller/management/customer/AddCustomerServlet.java]: Don't create a new `CustomerDAO`. Use `Singleton.customerDAO` instead.

File: src/main/java/io/hardingadonis/saledock/controller/management/customer/AddCustomerServlet.java [hunk 2]
```diff
@@ -1,20 +1,57 @@
...
+            String email = request.getParameter("email");
+
+            if (name != null && code != null && address != null && email != null) {
+                Customer customer = new Customer();
+                customer.setName(name);
+                customer.setCode(code);
+                customer.setAddress(address);
+                customer.setEmail(email);
+
+                boolean result = dao.insert(customer);
```
Comment (hardingadonis) [inline src/main/java/io/hardingadonis/saledock/controller/management/customer/AddCustomerServlet.java]: `CustomerDAO` has `save` method to insert and update a customer. Don't create new method.

File: src/main/java/io/hardingadonis/saledock/dao/IDAO.java [hunk 3]
```diff
@@ -13,4 +13,6 @@ public interface IDAO<T> {
     default Long count() {
         return 0L;
     }
+    
+    public boolean insert (T t);
```
Comment (hardingadonis) [inline src/main/java/io/hardingadonis/saledock/dao/IDAO.java]: Remove this method

File: src/test/java/TestCus.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+
```
Comment (hardingadonis) [inline src/test/java/TestCus.java]: Remove this file

**Review by thson58 — APPROVED** [review]

File: src/main/java/io/hardingadonis/saledock/controller/management/customer/AddCustomerServlet.java [hunk 5]
```diff
@@ -1,20 +1,56 @@
 package io.hardingadonis.saledock.controller.management.customer;
 
+
+import io.hardingadonis.saledock.model.Customer;
```
Comment (hardingadonis) [inline src/main/java/io/hardingadonis/saledock/controller/management/customer/AddCustomerServlet.java]: Use `import io.hardingadonis.saledock.model.*;`

File: src/main/java/io/hardingadonis/saledock/controller/management/customer/AddCustomerServlet.java [hunk 6]
```diff
@@ -1,20 +1,56 @@
...
+import io.hardingadonis.saledock.utils.Singleton;
```
Comment (hardingadonis) [inline src/main/java/io/hardingadonis/saledock/controller/management/customer/AddCustomerServlet.java]: Use `io.hardingadonis.saledock.utils.*;`

**Review by hardingadonis — APPROVED** [review]

**Review by bakaqc — APPROVED** [review]

**Pull request merged**